core/swift3Action/Dockerfile

```diff
@@ -4,16 +4,24 @@ FROM buildpack-deps:trusty
 
 ENV DEBIAN_FRONTEND noninteractive
 
-# Upgrade and install Python dependencies
+# Upgrade and install Python and helpers
 RUN apt-get -y purge && \
     apt-get -y update && \
-    apt-get -y install --fix-missing python2.7 python-distribute python-pip
+    apt-get -y install --fix-missing python2.7 && \
+    apt-get -y install --fix-missing python2.7-dev && \
+    apt-get -y install --fix-missing python-distribute && \
+    apt-get -y install --fix-missing python-pip && \
+    apt-get -y install --fix-missing python-gevent && \
+    apt-get clean && \
+    pip install -U flask
 
-# Install Python proxy support
-RUN apt-get -y install --fix-missing python2.7-dev python-gevent python-flask
+# Upgrade and install Swift dependencies.
+RUN apt-get -y dist-upgrade
 
-# Upgrade and install Swift dependencies
-RUN apt-get -y install build-essential wget clang libedit-dev libicu52 libxml2-dev
+# OpenWhisk dependencies
+RUN apt-get -y install  build-essential wget clang libedit-dev libicu52 rsync libxml2-dev emacs24-nox
```

Comment:
I think you should avoid changing the top part of this dockerfile altogether. I don't see a reason why these were changed. Can someone help me understand?